### PR TITLE
Create facade and reference assembly for mscorlib

### PR DESCRIPTION
**Pull request by AlexGhiondea**
Introduce reference assembly and facade for mscorlib.

This does not yet contain the folder renames but should unblock further work here.

### CODE REVIEW

**Comment by AlexGhiondea** [comment]
@dotnet-bot please test Ubuntu x64 Checked Build and Test

**Comment by AlexGhiondea** [comment]
@dotnet-bot please test OSX x64 Checked Build and Test

**Comment by gkhanna79** [comment]
LGTM. I will perform the switch of mscorlib impl/facade with my change.

**Comment by gkhanna79** [comment]
@AlexGhiondea I dont see the facade and ref folders getting generated. Your PR does not appear to hook up the build of the projects. Did you intentionally leave that out?

**Comment by AlexGhiondea** [comment]
@gkhanna79 yes --  to enable them for now we can add them to the src\build.proj file and remove the build for mscorlib.csproj.

I will create a build project for mscorlib when I move the sources and hook that up to the build.

**Comment by gkhanna79** [comment]
Okay - I will wireup facade generation as part of my changes and stop building mscorlib.dll

**Comment by AlexGhiondea** [comment]
@gkhanna79 this is what I had in mind (and removing the mscorlib implementation if that is ready): https://github.com/dotnet/coreclr/compare/master...AlexGhiondea:enableBuild?expand=1

**Comment by gkhanna79** [comment]
Do we still need to build mscorlib.dll as part of this change? I can switch over to Ref and Facade builds.

**Comment by AlexGhiondea** [comment]
In that case no, it can be removed.

**Comment by gkhanna79** [comment]
@AlexGhiondea Can you please look into the facade build break on non-Windows (use my branch from https://github.com/dotnet/coreclr/pull/4874)?

File: src/mscorlib/facade/mscorlib.csproj [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    <!-- Note the trailing '\\' - they are needed so that genfacade does not fail due to an invalid path
+         In particular, the facadepath is sent in quotes. 
+         And the IntermediateOutputPath is required to end in a '\' in the coreclr repo. This means that the 
+         facadepath ends up being sent as 'mypath\"' and that trips the Path normalization.
+         Introducing a second '\' tricks the normalization logic to treat this '\\"' as '\"' and not complain.    -->
+    <IntermediateOutputPath>$(BaseIntermediateOutputPath)\$(BuildOS).$(BuildArch).$(Configuration)\facade\\</IntermediateOutputPath>
+    <OutputPath>$(BinDir)\facade</OutputPath>
+  </PropertyGroup>
+
+  <ItemGroup Condition="'$(IsPartialFacadeAssembly)'=='true'">
```
Comment (weshaggard) [inline src/mscorlib/facade/mscorlib.csproj]: Are there any cases where IsPartialFacadeAssembly isn't true?

File: src/mscorlib/ref/mscorlib.csproj [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
+    
+    <!-- This prevents the default MsBuild targets from referencing System.Core.dll -->
+    <AddAdditionalExplicitAssemblyReferences>false</AddAdditionalExplicitAssemblyReferences>
+    <!-- These prevent the default MsBuild targets from referencing System.dll and mscorlib.dll -->
+    <NoStdLib>true</NoStdLib>
+    <NoCompilerStandardLib>true</NoCompilerStandardLib>
+    
+    <SubsystemVersion>6.00</SubsystemVersion>
+    <UTF8OutPut>true</UTF8OutPut>
```
Comment (weshaggard) [inline src/mscorlib/ref/mscorlib.csproj]: A reference assembly shouldn't need any of this. We should try and clean it up.

**Pull request merged**